src/main/java/marquez/dao/JobRunDAO.java

```diff
@@ -0,0 +1,81 @@
+package marquez.dao;
+
+import java.util.UUID;
+import marquez.core.models.JobRun;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.sqlobject.CreateSqlObject;
+import org.jdbi.v3.sqlobject.SqlObject;
+import org.jdbi.v3.sqlobject.config.RegisterRowMapper;
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RegisterRowMapper(JobRunRow.class)
+public interface JobRunDAO extends SqlObject {
+  Logger LOG = LoggerFactory.getLogger(JobRunDAO.class);
+
+  @CreateSqlObject
+  JobRunStateDAO createJobRunStateDAO();
+
+  default void insert(JobRun jobRun) {
+    try (final Handle handle = getHandle()) {
+      handle.useTransaction(
+          h -> {
+            h.createUpdate(
+                    "INSERT INTO job_runs (guid, started_at, job_version_guid, current_state, job_run_args_hex_digest) "
+                        + "VALUES (:guid, :startedAt, :jobVersionGuid, :currentState, :runArgsHexDigest)")
+                .bindBean(jobRun)
+                .execute();
+            createJobRunStateDAO()
+                .insert(UUID.randomUUID(), jobRun.getGuid(), jobRun.getCurrentState());
+          });
+    } catch (Exception e) {
+      // TODO: Add better error handling
+      LOG.error(e.getMessage());
+    }
+  }
+
+  default void update(JobRun jobRun) {
+    try (final Handle handle = getHandle()) {
+      handle.useTransaction(
+          h -> {
+            h.createUpdate(
+                    "UPDATE job_runs SET current_state = :currentState, started_at = :startedAt, ended_at = :endedAt where guid = :guid")
+                .bindBean(jobRun)
+                .execute();
+            createJobRunStateDAO()
+                .insert(UUID.randomUUID(), jobRun.getGuid(), jobRun.getCurrentState());
+          });
+    } catch (Exception e) {
+      // TODO: Add better error handling
+      LOG.error(e.getMessage());
+    }
+  }
+
+  String updateStateSQL =
```

Comment:
nit: Please consider using StringBuilder

Reply:
will probably use `+` for these SQL string concats for readability. but for other cases, agreed that `StringBuilder` can be a better option